spec/helpers/faker/puppet_module_remote_repo.rb

```diff
@@ -0,0 +1,130 @@
+require 'open3'
+
+require_relative '../faker'
+
+# Fake a remote git repository that hold a puppet module
+#
+# This module allow to fake a remote repositiory using:
+#  - a bare repo
+#  - a temporary cloned repo to operate on the remote before exposition to modulesync
+#
+# Note: This module need to have working_directory sets before using it
+module Faker
+  class PuppetModuleRemoteRepo
+    attr_reader :name
+    attr_reader :namespace
+
+    def initialize(name, namespace)
+      @name = name
+      @namespace = namespace
+    end
+
+    def populate
+      FileUtils.chdir(Faker.working_directory) do
+        run "git init --bare '#{bare_repo_dir}'"
+        run "git clone '#{bare_repo_dir}' '#{tmp_repo_dir}'"
```

Comment:
I think open3 supports arrays for commands so can't you use `run ['git', 'clone', bare_repo_dir, tmp_repo_dir]`? This is a repeated pattern so I won't comment on it everywhere.

Reply:
OK, applied using percent notation to prevent verbosity.